astropy/io/ascii/src/tokenizer.c

```diff
@@ -595,7 +596,7 @@ long str_to_long(tokenizer_t *self, char *str)
     char *tmp;
     long ret;
     errno = 0;
-    ret = strtol(str, &tmp, 0);
+    ret = strtol(str, &tmp, 10);
```

Comment:
this looks to be the base of the string->long change.  Presumably because that's now being taken care of by setting the character?  But it is a good question for @dhomeier : isn't this a subtle change in how long columns are interpreted?